dohclient.go

```diff
@@ -140,8 +140,11 @@ func (d *DoHClient) Resolve(q *dns.Msg, ci ClientInfo) (*dns.Msg, error) {
 
 	d.metrics.query.Add(1)
 
+	ctx, cancel := context.WithTimeout(context.Background(), d.opt.QueryTimeout)
+	defer cancel()
+
 	// Build a DoH request and execute it
-	req, err := d.buildRequest(msg)
+	req, err := d.buildRequest(msg, ctx)
```

Comment:
Would you mind making the context the first argument? That's very much a standard in Go. See https://pkg.go.dev/context
> The Context should be the first parameter, typically named ctx: